### PR TITLE
feat: contribution guideline updates

### PR DESCRIPTION
**Pull request by vedant-jain03**
#### Describe the changes you have made in this PR -

New contributions guidelines

Contributed by: @Kartikay304 @Palaksharma23 @Prerna-0202 @pulkit-30 @vedant-jain03 
Lead by: @tachyons @Shivansh2407 @ayan-biswas0412 

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: CONTRIBUTING.md
Comment (ayan-biswas0412) [inline CONTRIBUTING.md:12]: After a core team member updates the labels please insert the label link here also.

**Review by ayan-biswas0412 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (vedant-jain03) [reply CONTRIBUTING.md:12]: Sure

**Review by Kartikay304 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (ayan-biswas0412) [reply CONTRIBUTING.md:11]: what is WDYT ?

Comment (vedant-jain03) [reply CONTRIBUTING.md:11]: I guess "What do you think?" 😅

Comment (Kartikay304) [reply CONTRIBUTING.md:11]: 😅 yeah WDTY="What do you think?"

Comment (ayan-biswas0412) [reply CONTRIBUTING.md:11]: > 😅 yeah WDTY="What do you think?"

then leave it 😅 there will be many of like me who don't know this short forms

**Review by pulkit-30 — COMMENTED** [review]
We can also add labels for a duplicate to identify the duplicate issues and question for questions (some ask questions about features) and priority to differentiate issues on the basis of their priorities as some issues need to be fixed immediately (Priority: high) and some are not (Priority: low).

WDYT @vedant-jain03 @ayan-biswas0412

File: CONTRIBUTING.md
Comment (pulkit-30) [reply CONTRIBUTING.md:11]: + we can add `priority` and `question` labels.

**Comment by ayan-biswas0412** [comment]
> We can also add labels for a duplicate to identify the duplicate issues and question for questions (some ask questions about features) and priority to differentiate issues on the basis of their priorities as some issues need to be fixed immediately (Priority: high) and some are not (Priority: low).
> 
> WDYT @vedant-jain03 @ayan-biswas0412 

I think priority will be a good one

**Comment by pulkit-30** [comment]
> I think priority will be a good one

I was thinking of one more, ... some issues are just opened and they have less information about the issue and need more information to reproduce and check the issue. We can have a label to differentiate between these issues.

**Comment by Palaksharma23** [comment]
nice work @vedant-jain03 🚀😃
like @ayan-biswas0412 mentioned, I was also thinking we should insert the label link in the label name.

**Comment by ayan-biswas0412** [comment]
> > I think priority will be a good one
> 
> I was thinking of one more, ... some issues are just opened and they have less information about the issue and need more information to reproduce and check the issue. We can have a label to differentiate between these issues.

fo that comment section in the issue is enough I think, we should not make a lot of micro issues which will lead to over maintenance at near future

**Review by pulkit-30 — APPROVED** [review]
Rest looks good. Great work @vedant-jain03 👍

**Review by Prerna-0202 — APPROVED** [review]
Great work @vedant-jain03, Looks good to me👍

File: CONTRIBUTING.md
Comment (tachyons) [reply CONTRIBUTING.md:12]: What about using the term "maintainers" instead of core team in this document ?

File: CONTRIBUTING.md
Comment (tachyons) [inline CONTRIBUTING.md:55]: Also mention when to close a PR discussion as resolved, especially one generated by static code tools lilke codeclimate or hound

File: CONTRIBUTING.md
Comment (tachyons) [inline CONTRIBUTING.md:77]: Discussions should in discussions page, not on issues

File: CONTRIBUTING.md
Comment (vedant-jain03) [reply CONTRIBUTING.md:60]: Done

File: CONTRIBUTING.md
Comment (gr455) [inline CONTRIBUTING.md:26]: Usually `help wanted` is open to anyone who can help. Maybe we can keep it like that?

Comment (vedant-jain03) [reply CONTRIBUTING.md:26]: Should I remove it then?

Comment (gr455) [reply CONTRIBUTING.md:26]: Don't remove it. Just modify it to say
`help wanted` labelled issues signify that the contributor requires help with something specific in the issue and your help is very much appreciated

File: CONTRIBUTING.md
Comment (gr455) [inline CONTRIBUTING.md:50]:
```suggestion
5. Commit your changes and push it to your fork of the repository.
```
Local repo can be confused with repository on local machine.

Comment (vedant-jain03) [reply CONTRIBUTING.md]: Done

**Comment by gr455** [comment]
Just a small suggestion. Let me know what you all think about this.
Let's use the word `core team member` as less as possible. We don't want to people to get intimidated by any power difference. Let's just use `a community member` for places like
```diff
- A core-team member will review your PR
+ A community member will review your PR
```
Let me know your thoughts on this

File: CONTRIBUTING.md
Comment (Prerna-0202) [inline CONTRIBUTING.md:69]: Hey @vedant-jain03, I think by mistake you have written the 'communtiy' in place of 'community' ...

File: CONTRIBUTING.md
Comment (Kartikay304) [inline CONTRIBUTING.md:25]: I think instead of  priority less, medium and hard we can add label p-1, p-2 and p-3 respectively. If an issue/pr has multiple labels by using short-terms we can reduce clutter 🤔.

**Review by Kartikay304 — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md
Comment (ayan-biswas0412) [reply CONTRIBUTING.md:62]: yah it will be good i think

**Review by ayan-biswas0412 — APPROVED** [review]
Good work 🎉.

File: CONTRIBUTING.md
Comment (Kartikay304) [reply CONTRIBUTING.md:25]: > no, not a new contributor may not be well equipped with short terms like p-1,p-2 etc.
We can add full forms in label description.

**Review by pulkit-30 — APPROVED** [review]
Great Work 🎉

File: CONTRIBUTING.md
Comment (gr455) [inline CONTRIBUTING.md:79]: `- [awaiting-approval](https://github.com/CircuitVerse/CircuitVerse/labels/awaiting-approval) labeled PRs are meant to be waiting for other communtiy members.`

**Review by gr455 — CHANGES_REQUESTED** [review]
Can you replace `... community member` by `... a community member`, excuse me for being a nitpick  :sweat_smile: 

Also just address the Language Check github action comments. Then we're good to merge.

**Review by Palaksharma23 — APPROVED** [review]

**Review by Kartikay304 — APPROVED** [review]

**Comment by vedant-jain03** [comment]
> Can you replace `... community member` by `... a community member`, excuse me for being a nitpick sweat_smile
> 
> Also just address the Language Check github action comments. Then we're good to merge.

@gr455  I guess that what professionalism is meant to be 💯 
I have updated some points you requested/

File: CONTRIBUTING.md
Comment (tachyons) [inline CONTRIBUTING.md:73]: We need to standardise what to use for multi word tags, space, - , or _

File: CONTRIBUTING.md
Comment (tachyons) [inline CONTRIBUTING.md:9]: This seems missing in the new guide

File: CONTRIBUTING.md
Comment (vedant-jain03) [reply CONTRIBUTING.md:73]: Can we use '-'? @tachyons

File: CONTRIBUTING.md
Comment (vedant-jain03) [reply CONTRIBUTING.md:9]: Added in line 10 sir.

File: CONTRIBUTING.md
Comment (tachyons) [inline CONTRIBUTING.md:45]: It seems that financial contrinution section via open collective is removed, Can you add them back ?

Comment (vedant-jain03) [reply CONTRIBUTING.md:45]: Oh sorry, I forget to add it.
Added

**Pull request merged**
